contracts/StreamV1.sol

```diff
@@ -61,7 +102,59 @@ contract StreamV1 is Ownable {
         // update storage
         recipients = _recipients;
         shareBPS = _shareBPS;
+        fundDuration = _fundDuration;
+        // emit event
+        emit RecipientsUpdated(_recipients, _shareBPS, _fundDuration);
+    }
+
+    /* admin functions - reward program */
+
+    function _registerBonusToken(address rewardProgramAddress, address bonusToken)
+        external
+        onlyOwner
+    {
+        Aludel(rewardProgramAddress).registerBonusToken(bonusToken);
+        // emit event
+        emit ProgramRegisterBonusToken(rewardProgramAddress, bonusToken);
+    }
+
+    function _registerVaultFactory(address rewardProgramAddress, address factory)
+        external
+        onlyOwner
+    {
+        Aludel(rewardProgramAddress).registerVaultFactory(factory);
+        // emit event
+        emit ProgramRegisterVaultFactory(rewardProgramAddress, factory);
+    }
+
+    function _removeVaultFactory(address rewardProgramAddress, address factory) external onlyOwner {
+        Aludel(rewardProgramAddress).removeVaultFactory(factory);
+        // emit event
+        emit ProgramRemoveVaultFactory(rewardProgramAddress, factory);
+    }
+
+    function _rescueTokensFromRewardPool(
+        address rewardProgramAddress,
+        address token,
+        address recipient,
+        uint256 amount
+    ) external onlyOwner {
+        Aludel(rewardProgramAddress).rescueTokensFromRewardPool(token, recipient, amount);
+        // emit event
+        emit ProgramRescueTokensFromRewardPool(rewardProgramAddress, token, recipient, amount);
+    }
+
+    function _transferOwnership(address rewardProgramAddress, address newOwner) external onlyOwner {
+        Aludel(rewardProgramAddress).transferOwnership(newOwner);
+        // emit event
+        emit ProgramTransferOwnership(rewardProgramAddress, newOwner);
+    }
```

Comment:
imo unnecessary to have these functions - simpler to use the arbitrary external call